### PR TITLE
EXSWHTEC-28 - Implement tests for hipDeviceSetMemPool and hipDeviceGetMemPool

### PR DESCRIPTION
**Pull request by mirza-halilcevic**
- Negative parameter tests
- Default return value test
- Multithreaded test

Depends-on: #2887

### CODE REVIEW

File: tests/catch/unit/device/hipDeviceSetGetMemPool.cc [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+    }
+
+    void TestPart3() {
+      hipMemPool_t returned_mem_pool;
+      HIP_CHECK(hipDeviceGetMemPool(&returned_mem_pool, 0));
+
+      REQUIRE(returned_mem_pool == mem_pool_);
+    }
+
+    void TestPart4() { HIP_CHECK_THREAD(hipMemPoolDestroy(mem_pool_)); }
```
Comment (rorake) [inline tests/catch/unit/device/hipDeviceSetGetMemPool.cc:130]: Remove the empty lines between TestPart2/3/4

Comment (mirza-halilcevic) [reply tests/catch/unit/device/hipDeviceSetGetMemPool.cc:130]: The suggested change is implemented. Please proceed with the review.

**Review by rorake — DISMISSED** [review]

File: tests/catch/unit/device/hipDeviceSetGetMemPool.cc [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+THE SOFTWARE.
+*/
+
+#include <hip_test_common.hh>
+#include <hip/hip_runtime_api.h>
+#include <threaded_zig_zag_test.hh>
+
+#if HT_AMD
```
Comment (chrispaquot) [inline tests/catch/unit/device/hipDeviceSetGetMemPool.cc]: Please remove this and make the test skipped on NV for now.

**Review by chrispaquot — CHANGES_REQUESTED** [review]

File: tests/catch/unit/device/hipDeviceSetGetMemPool.cc (hunk 2 above)
Comment (mirza-halilcevic) [reply tests/catch/unit/device/hipDeviceSetGetMemPool.cc]: The latest version of HIP includes mem pool support for Nvidia so this guard is now removed.

**Review by chrispaquot — APPROVED** [review]

**Pull request merged**